### ISSUE TITLE
MDRNN doesn't train properly on carracing?

### ISSUE THREAD

**Issue by hanbrianlee** [ctallec/world-models#41]
train loss does go down but validation loss generally goes up.. is this the expected behavior or am I using wrong hyperparameters?

Thanks,

![image](https://user-images.githubusercontent.com/30082628/137207019-7984620d-852f-4df6-ae78-7614fbffae6f.png)

**Comment by DennisRotondi** [comment]
I think that there is a mistake in the implementation of gmm loss, it is possible to obtain negative values and this is clearly incorrect. Using a log-likelihood appoach seems to give better results, the following code works slightly changing what they provide:

        batch = batch.unsqueeze(-2) 
        normal_dist = Normal(mus, sigmas) 
        g_log_probs = normal_dist.log_prob(batch)  
        g_probs = logpi[:,:,:,None] + g_log_probs 
        p_loss = torch.sum(g_probs, dim=-2) 
        p_loss = p_loss.view(-1,p_loss.shape[-1]).sum(dim=0)
        loss = - p_loss.mean() if reduce else - p_loss
        return {'loss':loss}
